store/modules/data/mutations.js

```diff
@@ -24,8 +24,10 @@ export default {
         const new_well_values = new_values[well_idx];
         for (const metric_name in state.heatmap_values) {
           if (new_well_values[TWITCH.METRIC_IDS[metric_name]] !== undefined) {
-            state.heatmap_values[metric_name].data[well_idx] =
-              new_well_values[TWITCH.METRIC_IDS[metric_name]];
+            const temp = state.heatmap_values[metric_name].data[well_idx].concat(
```

Comment:
This still has the same issue with state.heatmap_values[metric_name].data[well_idx] being written twice. I think the cleanest way to do this while allowing vue to still recognize that the value updated would be something like this
```
const metric_values = state.heatmap_values[metric_name]
metric_values.data[well_idx] = metric_values.data[well_idx].concat(new_well_values[TWITCH.METRIC_IDS[metric_name]]).slice(-5)
```